BootloaderCommonPkg/Include/Library/BootloaderCommonLib.h

```diff
@@ -395,6 +395,30 @@ GetPlatformDataPtr (
   VOID
   );
 
+/**
+  Gets component entry from the flash map by partition.
+
+  This function will look for the component matching the input signature
+  in the flash map, if found, it will look for the component with back up
+  flag based on the backup partition parmeter and will return the
+  entry of the component from flash map.
+
+  @param[in]  Signature         Signature of the component information required
+  @param[in]  IsBackupPartition TRUE for Back up copy, FALSE for primary copy
+  @param[out] EntryDesc         Pointer to flash map entry
+
+  @retval    EFI_SUCCESS    Found the component with the matching signature.
+  @retval    EFI_NOT_FOUND  Component with the matching signature not found.
+
+**/
+EFI_STATUS
+EFIAPI
+GetComponentEntryByPartition (
+  IN  UINT32                Signature,

```

Comment:
Having 2 similar APIs GetComponentInfoByPartition  and GetComponentEntryByPartition. maybe we just replace GetComponentInfoByPartition  with new API?

Reply:
Let's decide to remove GetComponentInfoByPartition or not in a separate patch later.